src/modules/admin/Admin.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import csx from './Admin.scss';
+import AdminTabCategories, { AdminTabCategory } from './admin-tab-categories/AdminTabCategories';
+import TechnologiesTab from './technologies-tab/TechnologiesTab';
+import PatternsTab from './patterns-tab/PatternsTab';
+
+const Admin = () => {
+  const { category } = useParams();
+
+  const renderTabContent = () => {
```

Comment:
You should move this function before Admin component - this fn is created every time state changes